core/trino-main/src/main/java/io/trino/execution/scheduler/PipelinedStageExecution.java

```diff
@@ -495,6 +481,17 @@ private synchronized void noMoreSourceTasks(PlanFragmentId fragmentId)
         }
     }
 
+    private synchronized void updateSourceTasksOutputBuffers(Consumer<OutputBufferManager> updater)
+    {
+        for (PlanFragmentId sourceFragment : exchangeSources.keySet()) {
+            OutputBufferManager outputBufferManager = outputBufferManagers.get(sourceFragment);
+            updater.accept(outputBufferManager);
+            for (RemoteTask sourceTask : sourceTasks.get(sourceFragment)) {
+                sourceTask.setOutputBuffers(outputBufferManager.getOutputBuffers());
```

Comment:
This may do a remote call while within the synchronized block.

Reply:
The actual implementation doesn't. There are many call sites in the code base where the `RemoteTask` methods are called under a lock as well as the `RemoteTask` scheduling also happens under the lock. It's been like that for a very long time. But I agree that it is potentially dangerous. Do you think this issue can be addressed independently?

Reply:
No, it's ok. We can look at that separately. Just making sure it was intentional and consistent with how it used to work.